in-work/markov_chains.md

```diff
@@ -1515,9 +1523,10 @@ add to .bib
 ```{code-cell} ipython3
 def is_irreducible(P):
     k = P.shape[0]
+    result = np.zeros((k, k))
     for i in range(k):
-        P += np.linalg.matrix_power(P, i)
-    return np.all(P > 0)
+        result += np.linalg.matrix_power(P, i)
+    return np.all(result > 0)
```

Comment:
Please replace `k` with `n`, since we use `n` throughout as the lsize of the state space.